lib/routes/authorization.js

```diff
@@ -80,17 +81,19 @@ function generateCode(claims, client, scope, req) {
       },
       response: out
     });
+    activityEvent('generateCode', claims.uid, hex(req.payload.client_id), req.headers['user-agent']);
     return out;
   });
 }
 
-function generateGrant(claims, client, scope) {
+function generateGrant(claims, client, scope, req) {
   return db.generateToken({
     clientId: client.id,
     userId: buf(claims.uid),
     email: claims['fxa-verifiedEmail'],
     scope: scope
   }).then(function(token) {
+    activityEvent('generateGrant', claims.uid, hex(client.id), req.headers['user-agent']);
```

Comment:
Ok. Am just doing a rebase due to the changes in https://github.com/mozilla/fxa-oauth-server/pull/260 so a few things have moved.

From what I can tell, if I move it higher up, I don't yet have the 'claims' which I'm currently getting the uid from. I'll rebase and continue from there, there might be another way.